tavern_server/src/db_tables.sql

```diff
@@ -0,0 +1,676 @@
+	size			Size,
+	languages		TEXT,
+	race_type		RaceType
+);
+
+--Class tables
+CREATE TABLE Classes (
+	class_id		UUID,
+
+	name			TEXT,
+	hit_die			TEXT,
+	starting_wealth		TEXT,
+	bab_per_level		REAL,
+	skills_per_level	INT,
+	skills_attr		Attribute
+);
+
+CREATE TABLE Subclasses (
+	subclass_id		UUID,
+	class_id		UUID,
+
+	caster_type		CasterType,
+	casting_attr		Attribute
+);
+
+CREATE TABLE characterSubclasses (
+	char_id			UUID,
+	subclass_id		UUID,
+
+	levels_taken		INT,
+	hp_taken		INT,
+	skills_taken		INT
+);
+
+--Feats tables
+CREATE TABLE Feats (
+	feat_id			UUID,
+	req_id			UUID,
+
+	short_description	TEXT,
+	long_description	TEXT
+);
+
+CREATE TABLE SkillReqUnits (
+	req_id			UUID,
+	skill_unit_id		UUID
+);
+
+CREATE TABLE SkillFeatUnits (
+	skill_unit_id		UUID,
+	req_skill		Skill,
+	ranks			INT
+);
+
+CREATE TABLE AttributeReqUnits (
+	req_id			UUID,
+	attr_unit_id		UUID
+);
+
+CREATE TABLE AttributeFeatUnits (
+	attr_unit_id		UUID,
+	req_attr		Attribute,
+	score			INT
+);
+
+--this table creates a many to many realationship internally between feats.
+--any given feat can require a number of other feats to be taken,
+--and a given feat can be required by any number of other feats.
+--this table accomplishes that.
+CREATE TABLE RequiredFeats (
+	req_id			UUID,
+	feat__id		UUID
+);
+
+CREATE TABLE CharacterFeats (
+	char_id			UUID,
+	feat_id			UUID
+);
+
+CREATE TABLE RacialFeats (
+	race_id			UUID,
+	feat_id			UUID
+);
+
+CREATE TABLE ClassFeats (
+	class_id		UUID,
+	feat_id			UUID
+);
+
+--Features tables
+CREATE TABLE Features (
+	feature_id		UUID,
+	description		TEXT
+);
+
+CREATE TABLE ClassFeatures (
+	class_id		UUID,
+	feature_id		UUID,
+	is_default		BOOLEAN
+);
+
+CREATE TABLE CharacterFeatures (
+	char_id			UUID,
+	feature_id		UUID
+);
+
+CREATE TABLE SubclassFeatures (
+	subclass_id		UUID,
+	feature_id		UUID
+);
+
+--class proficiencies tables
+CREATE TABLE ClassProficientSingleWeapon (
+	class_id		UUID,
+	item_id			UUID
+);
+
+CREATE TABLE ClassProficientWeaponClass (
+	class_id		UUID,
+	weapon_class		WeaponClass
+);
+
+CREATE TABLE ClassProficientWeaponAllBut (
+	 class_id		UUID,
+	weapon_class		WeaponClass
+);
+
+CREATE TABLE ClassProficientArmorClass (
+	class_id		UUID,
+	armor_class		ArmorClass
+);
+
+CREATE TABLE ClassProficientArmorAllBut (
+	class_id		UUID,
+	armor_class		ArmorClass
+);
+
+--Spells tables
+CREATE TABLE Spells (
+	spell_id		UUID,
+
+	name			TEXT,
+	level			INT,
+	school			MagicSchool,
+
+	casting_time		INT,
+	range			SpellRange,
+	area			TEXT,
+	duration_per_level	INT,
+	saving_throw		SaveThrow,
+	spell_resistance	BOOLEAN,
+	description		TEXT
+);
+
+CREATE TABLE CharacterSpells (
+	char_id			UUID,
+	spell_id		UUID,
+	casts			INT
+);
+
+CREATE TABLE SubclassSpells (
+	class_id		UUID,
+	spell_id		UUID,
+	casts			INT,
+	req_level		INT
+);
+
+CREATE TABLE DomainSpells (
+	domain_id		UUID,
+	spell_id		UUID,
+	casts			INT
+);
+
+CREATE TABLE SpellComponents (
+	spell_id		UUID,
+	item_id			UUID,
```

Comment:
This looks like it takes into account only material components. What about voice and hand-waving? (Forgetting the correct terms for those atm)

Reply:
Implemented with a new field and ENUM called component_type and SpellComponent, respectively.